### PR TITLE
Write scripts for generating and seeding databases

### PR DESCRIPTION
**Pull request by zacharyhill**

### CODE REVIEW

**Comment by zacharyhill** [comment]
@rongsiu @swiimp can someone check out this PR for me please?

File: seed/generate-data.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      profile_photo_url: faker.image.avatar(),
+      rating: Number(((Math.random()) * 4).toFixed(0)) + 1,
+      relative_time_description: faker.date.recent(),
+      text: faker.lorem.paragraph(),
+      time: faker.date.past(),
+    };
+    const values = Object.values(record);
+    reviews.push(values.join(','));
+  }
+  if (reviews.length > 0) {
```
Comment (swiimp) [inline seed/generate-data.js]: These return statements could probably be refactored into a single ternary expression.

File: seed/generate-data.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+};
+
+const writeData = (writer, createRecord, callback, numOfRecords, encoding = 'utf8') => {
+  let i = 0;
+
+  const write = () => {
+    let ok = true;
+    do {
+      let data = createRecord(i, numOfRecords);
+      if (i !== numOfRecords - 1) {
```
Comment (swiimp) [inline seed/generate-data.js]: Could be refactored into a single Boolean expression

File: seed/generate-data.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    });
+  }
+  return JSON.stringify(business);
+};
+
+const writeData = (writer, createRecord, callback, numOfRecords, encoding = 'utf8') => {
+  let i = 0;
+
+  const write = () => {
+    let ok = true;
```
Comment (swiimp) [inline seed/generate-data.js]: Should be renamed to something more descriptive

File: seed/generate-data.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      generateReviews(reviewsFile, numOfRecords);
+    } else if (process.argv[3] === 'reviews') { // generate csv file for reviews
+      const reviewsFile = process.argv[4] || './reviews-data.csv';
+      const numOfRecords = process.argv[5] || 10e6;
+      generateReviews(reviewsFile, numOfRecords);
+    } else if (process.argv[3] === 'stores') { // generate csv file for stores
+      const reviewsFile = process.argv[4] || './stores-data.csv';
+      const numOfRecords = process.argv[5] || 10e6;
+      generateStores(reviewsFile, numOfRecords);
+    }
```
Comment (swiimp) [inline seed/generate-data.js]: There should probably be a default case here in case someone wants to use the script directly from the terminal

Comment (zacharyhill) [reply seed/generate-data.js]: This was by design. I do not want someone running this script without explicitly stating what they want to change as running this script on a production machine would have a lot of consequences. If someone wishes to run the script directly from the terminal they can add flags to specify non-defaults or they can simply run the bash script if they want the default generation script.

File: seed/generate-data.js (hunk 1 above)
Comment (zacharyhill) [reply seed/generate-data.js]: You can't return within a ternary operator.

**Pull request merged**